### PR TITLE
Fix incorrect error message (trivial)

### PR DESCRIPTION
**Pull request by ran-eh**
resolves #

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [ ] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Review by jtcohen6 — COMMENTED** [review]
Thanks for the PR @ran-eh! I'm not opposed to changing this error message in the spirit of making it clearer. I am curious to hear, in what way do you feel that the current language is incorrect?

Because `alert_non_existence` adjudicates each of (potentially many) selection criteria, it's possible that one criterion does not match any nodes while another does. To my mind, the phrase `nothing to do` is a bit confusing in such a case:
```bash
$ dbt run -m fct_sprints asdfgh
...
The selector 'asdfgh' does not match any nodes - nothing to do

14:21:31 | Concurrency: 8 threads (target='dev')
14:21:31 |
14:21:31 | 1 of 1 START table model dbt_jcohen.fct_sprints...................... [RUN]
```

Perhaps we could simplify this message to just: `The selection criterion 'asdfgh' does not match any nodes`. (While we're here, we may as well disambiguate from YAML `selectors` / `--selector`, too.)

**Comment by ran-eh** [comment]
> Thanks for the PR @ran-eh! I'm not opposed to changing this error message in the spirit of making it clearer. I am curious to hear, in what way do you feel that the current language is incorrect?
> 
> Because `alert_non_existence` adjudicates each of (potentially many) selection criteria, it's possible that one criterion does not match any nodes while another does. To my mind, the phrase `nothing to do` is a bit confusing in such a case:
> 
> ```shell
> $ dbt run -m fct_sprints asdfgh
> ...
> The selector 'asdfgh' does not match any nodes - nothing to do
> 
> 14:21:31 | Concurrency: 8 threads (target='dev')
> 14:21:31 |
> 14:21:31 | 1 of 1 START table model dbt_jcohen.fct_sprints...................... [RUN]
> ```
> 
> Perhaps we could simplify this message to just: `The selection criterion 'asdfgh' does not match any nodes`. (While we're here, we may as well disambiguate from YAML `selectors` / `--selector`, too.)

Agreed and changed.

**Review by jtcohen6 — APPROVED** [review]
Thanks for this @ran-eh!

File: core/dbt/graph/selector.py [hunk 1]
```diff
@@ -25,8 +25,7 @@ def get_package_names(nodes):
 def alert_non_existence(raw_spec, nodes):
     if len(nodes) == 0:
         warn_or_error(
-            f"The selector '{str(raw_spec)}' does not match any nodes and will"
-            f" be ignored"
+            f"The selection criterion '{str(raw_spec)}' does not match any nodes"
```
Comment (jtcohen6) [inline core/dbt/graph/selector.py:28]: `flake8` just takes issue with this line being two characters too long:
```
core/dbt/graph/selector.py:28:80: E501 line too long (81 > 79 characters)
```
So perhaps:
```suggestion
            f"The selection criterion '{str(raw_spec)}' does not match"
            f" any nodes"
```

**Review by jtcohen6 — CHANGES_REQUESTED** [review]
Whoops, sorry for the over-eager approval. One tiny change and then this should be fit to ship.

**Review by jtcohen6 — APPROVED** [review]

**Pull request merged**